jlm/rvsdg/gamma.cpp

```diff
@@ -344,6 +346,56 @@ GammaNode::MapBranchResultExitVar(const rvsdg::input & input) const
   return ExitVar{ std::move(branchResults), Node::output(input.index()) };
 }
 
+void
+GammaNode::RemoveExitVars(const std::vector<ExitVar> & exitvars)
+{
+  std::vector<std::size_t> indices;
+  for (const auto & exitvar : exitvars)
+  {
+    indices.push_back(exitvar.output->index());
+  }
+  std::sort(
+      indices.begin(),
+      indices.end(),
+      [](std::size_t x, std::size_t y)
+      {
+        return x > y;
+      });
+  for (std::size_t index : indices)
+  {
+    for (std::size_t r = 0; r < nsubregions(); ++r)
+    {
+      subregion(r)->RemoveResult(index);
+    }
+    RemoveOutput(index);
+  }
```

Comment:
I understand that the indices are sorted to avoid shifting that invalidates later indices, but we could also avoid the O(N*M) removal here, and replace it with O(N+M), if we do a single sweep.

Reply:
I think it needs to be O(#vars_removed * #regions), that's what the inner loop amounts to. So I am not entirely certain what you mean by "single sweep" here?